lib/src/Uncertainty/Distribution/JointDistribution.cxx

```diff
@@ -81,14 +81,14 @@ JointDistribution::JointDistribution(const DistributionCollection & coll)
 
 /* Default constructor */
 JointDistribution::JointDistribution(const DistributionCollection & coll,
-    const Distribution & copula)
+                                     const Distribution & kernel)
   : DistributionImplementation()
   , distributionCollection_()
-  , copula_(copula)
+  , kernel_(kernel)
 {
   setName("JointDistribution");
-  if (!copula.isCopula()) throw InvalidArgumentException(HERE) << "Error: the given distribution=" << copula << " is not a copula.";
-  setDimension(copula.getDimension());
+  if (!(kernel.getRange() == Interval(kernel.getDimension()))) throw InvalidArgumentException(HERE) << "Error: the given kernel=" << kernel << " has a range=" << kernel.getRange() << " different from the unit hypercube.";
```

Comment:
Perhaps it is too late for that but I would propose the name "pseudo-copula" or "semi-copula" instead of "kernel". IMO it is better than the (too) generic "kernel" because it tells that this object shares some properties with a copula (ie is a multi-dimensional distribution on the unit cube).

Reply:
No. Semi-copula is another concept (see the book of Nelsen, which is full of semi-copulas), and pseudo-copula is ugly. I don't want to invent a copula-related concept in the hurry of a RC just to tell that any distribution taking values in [0,1]^d can work. BTW:
  * The dimension can be equal to 1, and it is precisely the case when taking 1D margins of such distributions in the non-copula case
  * The range doesn't have to be exactly the unit cube but has to be included into it for the maths to work. A kind of truncation in the dependence part of the distribution. I will change the test accordingly.

Reply:
Well, we might need to change it later if we come up with a name we can agree on. I will attempt just one additional name suggestion: "unitCubeDistribution_" (it still works even if the support only needs to be included in the unit cube).